File: Code Generation/responses/cpp/claude/medium/zero-array-transformation-iv.cpp
#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
    int minZeroArray(vector<int> nums, vector<vector<int>> queries) {
        int n = nums.size();
        int q = queries.size();
        
        auto canZero = [&](int k) -> bool {
            vector<long long> diff(n + 1, 0);
            for (int i = 0; i < k; i++) {
                int l = queries[i][0], r = queries[i][1], val = queries[i][2];
                diff[l] += val;
                diff[r + 1] -= val;
            }
            long long cur = 0;
            for (int i = 0; i < n; i++) {
                cur += diff[i];
                if (cur < nums[i]) return false;
            }
            return true;
        };
        
        if (canZero(0)) return 0;
        
        int lo = 1, hi = q, ans = -1;
        while (lo <= hi) {
            int mid = (lo + hi) / 2;
            if (canZero(mid)) {
                ans = mid;
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return ans;
    }
};

File: Code Generation/responses/python/claude/medium/zero-array-transformation-iv.py
from typing import List

class Solution:
    def minZeroArray(self, nums: List[int], queries: List[List[int]]) -> int:
        n = len(nums)
        
        def can_zero(k):
            # Use difference array to track max decrements possible at each index
            diff = [0] * (n + 1)
            for i in range(k):
                l, r, val = queries[i]
                diff[l] += val
                diff[r + 1] -= val
            
            curr = 0
            for i in range(n):
                curr += diff[i]
                if curr < nums[i]:
                    return False
            return True
        
        if can_zero(0):
            return 0
        
        lo, hi = 1, len(queries)
        if not can_zero(hi):
            return -1
        
        while lo < hi:
            mid = (lo + hi) // 2
            if can_zero(mid):
                hi = mid
            else:
                lo = mid + 1
        
        return lo